src/main/java/Main.java

```diff
@@ -294,6 +299,19 @@ private static void _main(String[] args) throws Exception {
         mainMethod.invoke(null,new Object[]{arguments.toArray(new String[0])});
     }
 
+    /**
+     * reads up to maxRead bytes from InputStream if available into a String
+     *
+     * @param in
+     * @param maxToRead
+     * @return
+     * @throws IOException
+     */
+    private static String readStringNonBlocking(InputStream in, int maxToRead) throws IOException {
+        byte [] buffer = new byte[Math.min(in.available(), maxToRead)];
```

Comment:
I would throw IOException if the entire string cannot be read.

Reply:
When Jenkins starts up it only reads data that is already in stdin(result of piping). This way it does not block thus allowing to run war without passing any arguments just like it does today. This eliminates the need for --paramsFromStdIn you proposed(and I initially considered as well).